### PR TITLE
expression: fix the wrong behaviour bug of vec LeastTime

### LINKED ISSUE #17826: Wrong behaviour of MySQL function LEAST()
### 1. Minimal reproduce step (Required)

Use [copr-test](https://github.com/tikv/copr-test) to run `make no-push-down`. Then type this sql to make a query:

```sql
select col_char_255, col_date, least(col_char_255, col_date) from table1000_int_autoinc where least(col_char_255, col_date) and col_date is null;
```

The results will be like:

```
+--------------+----------+-------------------------------+
| col_char_255 | col_date | least(col_char_255, col_date) |
+--------------+----------+-------------------------------+
| 03:55:21     | NULL     | 03:55:21                      |
| -2008481792  | NULL     | -2008481792                   |
| -269615104   | NULL     | -269615104                    |
| 1074462720   | NULL     | 1074462720                    |
| -529334272   | NULL     | -529334272                    |
| 16:11:33     | NULL     | 16:11:33                      |
| 17:44:01     | NULL     | 17:44:01                      |
| 15:41:44     | NULL     | 15:41:44                      |
+--------------+----------+-------------------------------+
8 rows in set, 102 warnings (0.04 sec)
```

### 2. What did you expect to see? (Required)

According to the [MySQL 8.0 Reference Manual of LEAST()](https://dev.mysql.com/doc/refman/8.0/en/comparison-operators.html#function_least), if any argument is NULL, the result is NULL. No comparison is needed. So the results of `least(col_char_255, col_date)` should all be NULL.

### 3. What did you see instead (Required)

You can see the `least(col_char_255, col_date)`  results are all not NULL.

### 4. Affected version (Required)

```
+ TiDB version:
Release Version: v4.0.0-beta.2-572-g09af066c5-dirty
Edition: Community
Git Commit Hash: 09af066c546cdd2ce4f9cf3af354345244ee706b
Git Branch: master
UTC Build Time: 2020-06-06 01:17:25
GoVersion: go1.14.3
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
```

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

Issue Number: close #17826 

Problem Summary:

LEAST() function's behaviour is inconsistent with MySQL.

### What is changed and how it works?

What's Changed:

* vectorized LeastTime function

How it Works:

* Make a deep copy when find an invalid time.
* Return NULL if any argument is NULL.

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

No release note

### CODE REVIEW

**Comment by JmPotato** [comment]
@sre-bot /run-all-tests

**Comment by JmPotato** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/133 tikv=pr/7888

**Comment by JmPotato** [comment]
> Thanks for your contribution!!
> Should the row-based implementation `builtinLeastTimeSig.evalString()` also be changed?
> And please add some tests for this in `expression/builtin_compare.go`.

The row-based implementation is not affected by this bug. And I added some unit tests to cover the invalid time and NULL cases. PTAL, thx!

File: expression/builtin_compare_vec.go [hunk 1]
```diff
@@ -742,7 +742,13 @@ func (b *builtinLeastTimeSig) vecEvalString(input *chunk.Chunk, result *chunk.Co
...
 					return err
 				} else if !findInvalidTime[i] {
-					invalidValue[i] = result.GetString(i)
+					// Make a deep copy here.
+					// Otherwise invalidValue will internally change with result.
+					rowContent := result.GetString(i)
+					invalidValueTmp := make([]byte, len(rowContent))
+					copy(invalidValueTmp, rowContent)
+
+					invalidValue[i] = string(invalidValueTmp)
```
Comment (qw4990) [inline expression/builtin_compare_vec.go:751]: How about:
```
invalidValue[i] = string(result.GetBytes(i))
```

**Comment by JmPotato** [comment]
@sre-bot /run-all-tests

**Comment by JmPotato** [comment]
What's wrong with the CI test?

**Comment by JmPotato** [comment]
@sre-bot /run-all-tests

**Comment by JmPotato** [comment]
Some of the CI tests keep failing randomly. Is this normal?

**Comment by JmPotato** [comment]
@sre-bot /run-all-tests

**Pull request merged**